### DISCUSSION TITLE
Pagination on large collections?

### DISCUSSION THREAD

**Discussion by stevebrowndotco** [refinedev/refine#5553] [General]
Hello! I have been testing the inferencer with the simple rest API integration. I noticed that pagination is figured out by first querying the entire list in the rest query, and then by making subsequent calls with the start and end params.

What about large collections where it is impractical to query the entire database? I'm talking thousands of records. Can the inferencer look for other properties in the API like a meta property colacted next to the data to get the total for pagination? Is it supported in the simple rest API integration, or do I need to "swizzle" it. Are there any examples for this?

**Comment by alicanerdurmaz** [comment]
Hello @stevebrowndotco 

`inferencer` is designed to speed up the development process, we do not recommend using it in production.

You can copy the generated code and edit it according to your needs.

**Reply by stevebrowndotco** [reply]
Hello,

Yes I understand that it is not used for production, but it's purpose is to generate the code as a basis right?

Anyway, I found the answer to what I am looking for, the simple rest API gets the pagination total by checking the X-Total-Count response header. I couldn't find this in the docs. Also I was wrong in my first message about it checking every single record, it doesnt.
